File: launchdarkly/java/src/main/java/oseg/launchdarkly_examples/PostHoldoutExample.java
package oseg.launchdarkly_examples;

import com.launchdarkly.client.ApiException;
import com.launchdarkly.client.Configuration;
import com.launchdarkly.client.api.*;
import com.launchdarkly.client.auth.*;
import com.launchdarkly.client.JSON;
import com.launchdarkly.client.model.*;

import java.io.File;
import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.ArrayList;
import java.util.List;
import java.util.Map;

public class PostHoldoutExample
{
    public static void main(String[] args)
    {
        var config = Configuration.getDefaultApiClient();
        ((ApiKeyAuth) config.getAuthentication("ApiKey")).setApiKey("YOUR_API_KEY");

        var metrics1 = new MetricInput();
        metrics1.key("metric-key-123abc");
        metrics1.isGroup(true);
        metrics1.primary(true);

        var metrics = new ArrayList<MetricInput>(List.of (
            metrics1
        ));

        var holdoutPostRequest = new HoldoutPostRequest();
        holdoutPostRequest.name("holdout-one-name");
        holdoutPostRequest.key("holdout-key");
        holdoutPostRequest.description("My holdout-one description");
        holdoutPostRequest.randomizationunit("user");
        holdoutPostRequest.holdoutamount("10");
        holdoutPostRequest.primarymetrickey("metric-key-123abc");
        holdoutPostRequest.prerequisiteflagkey("flag-key-123abc");
        holdoutPostRequest.attributes(List.of (
            "country",
            "device",
            "os"
        ));
        holdoutPostRequest.metrics(metrics);

        try
        {
            var response = new HoldoutsBetaApi(config).postHoldout(
                "projectKey_string", // projectKey
                "environmentKey_string", // environmentKey
                holdoutPostRequest
            );

            System.out.println(response);
        } catch (ApiException e) {
            System.err.println("Exception when calling HoldoutsBetaApi#postHoldout");
            System.err.println("Status code: " + e.getCode());
            System.err.println("Reason: " + e.getResponseBody());
            System.err.println("Response headers: " + e.getResponseHeaders());
            e.printStackTrace();
        }
    }
}

File: launchdarkly/kotlin/src/main/kotlin/oseg/launchdarkly_examples/PostHoldoutExample.kt
package oseg.launchdarkly_examples

import com.launchdarkly.client.infrastructure.*
import com.launchdarkly.client.apis.*
import com.launchdarkly.client.models.*

import java.io.File
import java.time.LocalDate
import java.time.OffsetDateTime
import kotlin.collections.ArrayList
import kotlin.collections.List
import kotlin.collections.Map
import com.squareup.moshi.adapter

@ExperimentalStdlibApi
class PostHoldoutExample
{
    fun postHoldout()
    {
        ApiClient.apiKey["Authorization"] = "YOUR_API_KEY"

        val metrics1 = MetricInput(
            key = "metric-key-123abc",
            isGroup = true,
            primary = true,
        )

        val metrics = arrayListOf<MetricInput>(
            metrics1,
        )

        val holdoutPostRequest = HoldoutPostRequest(
            name = "holdout-one-name",
            key = "holdout-key",
            description = "My holdout-one description",
            randomizationunit = "user",
            holdoutamount = "10",
            primarymetrickey = "metric-key-123abc",
            prerequisiteflagkey = "flag-key-123abc",
            attributes = listOf (
                "country",
                "device",
                "os",
            ),
            metrics = metrics,
        )

        try
        {
            val response = HoldoutsBetaApi().postHoldout(
                projectKey = "projectKey_string",
                environmentKey = "environmentKey_string",
                holdoutPostRequest = holdoutPostRequest,
            )

            println(response)
        } catch (e: ClientException) {
            println("4xx response calling HoldoutsBetaApi#postHoldout")
            e.printStackTrace()
        } catch (e: ServerException) {
            println("5xx response calling HoldoutsBetaApi#postHoldout")
            e.printStackTrace()
        }
    }
}
